agent/api/transitiondependency.go

```diff
@@ -40,6 +45,15 @@ type ContainerDependency struct {
 	DependentStatus ContainerStatus `json:"DependentStatus,omitempty"`
 }
 
+// ResourceDependency defines the relationship between a dependent container
+// and its resource dependency.
+type ResourceDependency struct {
+	// ResourceName defines the Resource on which a transition depends
+	ResourceName string `json:"ResourceName"`
+	// SatisfiedStatus defines the status that satisfies the dependency
+	SatisfiedStatus taskresource.ResourceStatus `json:"SatisfiedStatus"`
```

Comment:
It'd be better to add a getter for this and use that method instead so that there's some flexibility in the future to potentially tweak that logic without changing code at a lot of places. something like `RequiredStatus() taskresource.ResourceStatus`. Also, `RequiredStatus` sounds like a better name.